---
title: 'Predict T cell dysfunction genes from TCGA Pancreatic Cancer Cohort'
author: 'Yufan Zhou'
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: TRUE
    depth: 3
editor_options: 
  chunk_output_type: console
---


***
### Overview
Adoptive cell therapy with chimeric antigen receptor expressing T (CAR T) cells has shown promising therapeutic efficacy against leukemia and lymphoma. However, CAR T cells in solid tumors fail to be as effective as in liquid tumors since they enter into a hyporesponsive (exhausted or dysfunctional) state that is induced by chronic antigen stimulation in cancer. To determine what kind of genes contribute to T cell exhaustion/ dysfunction from clinical patients sample would be meaningful to guide next-generation CAR T cell production. I identitified around two hundreds potential signatures of T cell dysfunction from pancreatic tumor cohort by testing how the expression of each gene in tumor interacts with the classiscal T cell specific dysfunction biomarkers. These potential T cell dysfunction genes were highly correlated with T cell activation and was correlated with poor survival rate. 


### Introduction 
Chimeric antigen receptor (CAR) T cells have been genetically engineered to express a receptor that recognizes a specific antigen, have given rise to breakthroughs in treating hematological malignancies. However, their success in treating solid tumors has been limited. T cells that enter solid tumors can stop working due to a phenomenon called T cell dysfunction. T cell dysfunction is a hallmark of many cancers. Identifying and overcoming mechanisms associated with dysfunction in CAR T cells is of vital importance to generating CAR T cells that can proliferate and successfully eliminate tumor cells.

Gene expression biomarkers in T cells, such as PDCD1, HACVR2, LAG3, TOX, EOMES, CTLA4, CD8A have been demonstrated in mouse model to predict CAR T treatment benefits. However, current CAR T clinical trials have gene expression profiles on only a small number of samples, which are insufficient to uncover prognostic biomarkers. Alternatively, there are many public tumor profiling datasets from human clinical trials. Some specimens have high level of infiltration by cytotoxic T cells and poor survival rate, and T cells in these specimens tend to be in a dysfunctional state. Analyzing TCGA dataset might uncover uncharacterized T cell dysfunction genes. Identification of reproducible biomarkers that can be applied to enhance CAR T cell persistance.

### Methods

1. Correlation analysis

TCGA genomic data were downloaded from the UCSC cancer genome browser project. Gene expression was measured experimentally using the Illumina HiSeq 2000 RNA Sequencing platform and log2(x + 1) transformed. Spearman’s rank correlation test will be used to assess the strength of the relationship between each classical T cell biomarkers with rest of 50,000 genes. The strength of a correlation was measured by the correlation coefficient r. Correlation greater than 0.5 was used as criteria to select candidate gene. Biomarkers correlation gene sets were performed overlapping analysis. 

2. Survival analysis

Kaplan–Meier curves were constructed by dividing specimens at the one standard deviation above or below average expression of each selected  gene. Overall survival (OS) data were available for all 222 patients.

3. Differential gene expression

Differential gene expression was performed with Limma. Limma-voom was used to identify transcripts that were significantly differntially expressed between experimental groups using adjusted P value of <0.05.

4. GO and KEGG analysis

Gene ontology (http://www.geneontology.org/) provides a controlled vocabulary for describing biological processes (BP ontology), molecular functions (MF ontology) and cellular components (CC ontology). KEGG mapping is the process to map genes to molecular interaction/reaction/relation networks. 


***

This report shows the process of discovering potential T cell dysfunction genes from TCGA cohort, including:

* Data preprocess
* Correlation analysis
* Differential gene expression analysis
* GO and KEGG Analysis
* Survival analysis

Load the necessary libraries.

```{r lib, eval=T, echo=T, message=F, warning=F}
library(knitr)
library(pander)
library(biomaRt)
library(curl)
library(VennDiagram)
library(stringr)
library(edgeR)
library(RColorBrewer)
library(biomaRt)
library(DOSE)
library(org.Hs.eg.db)
library(topGO)
library(clusterProfiler)
library(pathview)
library(curl)
library(pheatmap)
library(RColorBrewer)
library(ggplot2)
library(tidyverse)
library(survival)
library(survminer)
library(viridis)

```

## Data Preprocess

**Datasets:**

* Transformed FPKM normalized count (n=182) 

* Transformed RSEM normalized count (n=183)

* Survival data (n=222) 

* Phenotype data (n=222)

Dataset were downloaded from GDC TCGA Pancreatic Cancer cohort: https://xenabrowser.net/datapages/

All gene expression data was log2(x+1) transformed




**Biomarkers:**

T cell dysfunction biomarkers: **CD8A, HAVCR2, LAG3, PDCD1, EOMES, TBX21,CTLA4** have been demonstrated in pre-clinical model to predict T cell dysfunction



## Correlation Analysis

### Load Dataset
```{r load_data, eval=T, echo=T}
data <- read.table(gzfile("./Data/TCGA-PAAD.htseq_fpkm.tsv.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)
data[1:6, 1:10]

# remove decimal of Ensembl_ID
data_modified <- sapply(strsplit(data$Ensembl_ID,"\\."), function(x) x[1])
data_new <- cbind(data_modified, data)
data_new <- data_new[, -2]

library(dplyr)
data2 <- data_new %>% dplyr::rename(ensembl_gene_id = data_modified)
```

### Annotation
Use ensembl gene id to annotate gene name.

Remove NA gene.

Simplify sample name to arabic numerals "1,2,3..182".

```{r anno, eval=T, echo=F}
# Use ensembl gene id to annotate
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
gene <-getBM(attributes=c("ensembl_gene_id","external_gene_name"), values =data2$ensembl_gene_id, mart = mart)
id <- match(data2$ensembl_gene_id , gene$ensembl_gene_id)
data2$Symbol <- gene$external_gene_name[id]

#  Remove missing value
data2 <- data2[complete.cases(data2[,184]),]

#  Simplify sample name
library(dplyr)
data.clean <- cbind(symbol = data2$Symbol, data2)
data.clean <- subset(data.clean, select=-c(ensembl_gene_id,Symbol))
colnames(data.clean) <- c("Symbol",seq(1,182))
```

```{r annotation_table, eval=T, warning=F,results="asis",echo=F}
pandoc.table(data.clean[1:6, 1:10], split.tables=Inf, caption="Parts of annotated dataframe")
```

### Correlation Analysis

Spearman’s rank correlation test will be used to assess the strength of the relationship between each classical T cell biomarkers with rest of 60,000 genes.

**Biomarker CD8A, HAVCR2, LAG3, PDCD1, EOMES, TBX21, CTLA4 will be tested repectively.**

1. HAVCR2
```{r correlation_havcr2, eval=F, echo=T}

# set an empty dataframe
result_havcr2 <- data.frame((matrix(nrow = 56553, ncol = 2)))
colnames(result_havcr2) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "HAVCR2")[-1]))
colnames(x) <- subset(data.clean, Symbol == "HAVCR2")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_havcr2 <- cor.test(x$HAVCR2, y[1:182,1], method = "spearman", exact=F)
  result_havcr2[i,2] <- res_havcr2$estimate
  result_havcr2[i,1] <- as.character(data.clean[i,1])
}
result_havcr2
write.csv(result_havcr2, file = "./correlation_result/result_havcr2.csv")
```
2. PDCD1
```{r correlation_pdcd1, eval=F, echo=T}
# set an empty dataframe
result_pdcd1 <- data.frame((matrix(nrow = 56553, ncol = 2)))
colnames(result_pdcd1) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "PDCD1")[-1]))
colnames(x) <- subset(data.clean, Symbol == "PDCD1")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_pdcd1 <- cor.test(x$PDCD1, y[1:182,1], method = "spearman", exact=F)
  result_pdcd1[i,2] <- res_pdcd1$estimate
  result_pdcd1[i,1] <- as.character(data.clean[i,1])
}
result_pdcd1
write.csv(result_pdcd1, file = "./correlation_result/result_pdcd1.csv")
```
3. LAG3
```{r correlation_lag3, eval=F, echo=T}

# set an empty dataframe
result_lag3 <- data.frame((matrix(nrow = 56553, ncol = 2)))
colnames(result_lag3) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "LAG3")[-1]))
colnames(x) <- subset(data.clean, Symbol == "LAG3")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_lag3 <- cor.test(x$LAG3, y[1:182,1], method = "spearman", exact=F)
  result_lag3[i,2] <- res_lag3$estimate
  result_lag3[i,1] <- as.character(data.clean[i,1])
}
result_lag3
write.csv(result_lag3, file = "./correlation_result/result_lag3.csv")
```
4. TBX21
```{r correlation_tbx21, eval=F, echo=T}

# set an empty dataframe
result_tbx21 <- data.frame((matrix(nrow = 56553, ncol = 2)))
colnames(result_tbx21) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "TBX21")[-1]))
colnames(x) <- subset(data.clean, Symbol == "TBX21")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_tbx21 <- cor.test(x$TBX21, y[1:182,1], method = "spearman", exact=F)
  result_tbx21[i,2] <- res_tbx21$estimate
  result_tbx21[i,1] <- as.character(data.clean[i,1])
}
result_tbx21
write.csv(result_tbx21, file = "./correlation_result/result_tbx21.csv")
```
5. EOMES
```{r correlation_eomes, eval=F, echo=T}

# set an empty dataframe
result_eomes <- data.frame((matrix(nrow = 56553, ncol = 2)))
colnames(result_eomes) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "EOMES")[-1]))
colnames(x) <- subset(data.clean, Symbol == "EOMES")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_eomes <- cor.test(x$EOMES, y[1:182,1], method = "spearman", exact=F)
  result_eomes[i,2] <- res_eomes$estimate
  result_eomes[i,1] <- as.character(data.clean[i,1])
}
result_eomes
write.csv(result_eomes, file = "./correlation_result/result_eomes.csv")
```
6. CTLA4
```{r correlation_ctla4, eval=F, echo=T}
# set an empty dataframe
result_ctla4 <- data.frame((matrix(nrow = 56553, ncol = 2)))
colnames(result_ctla4) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "CTLA4")[-1]))
colnames(x) <- subset(data.clean, Symbol == "CTLA4")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_ctla4 <- cor.test(x$CTLA4, y[1:182,1], method = "spearman", exact=F)
  result_ctla4[i,2] <- res_ctla4$estimate
  result_ctla4[i,1] <- as.character(data.clean[i,1])
}
result_ctla4
write.csv(result_ctla4, file = "./correlation_result/result_ctla4.csv")
```
7. CD8A
```{r correlation_cd8a, eval=F, echo=T}
# set an empty dataframe
result_cd8a <- data.frame((matrix(nrow = 56553, ncol = 2)))
colnames(result_cd8a) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "CD8A")[-1]))
colnames(x) <- subset(data.clean, Symbol == "CD8A")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_cd8a <- cor.test(x$CD8A, y[1:182,1], method = "spearman", exact=F)
  result_cd8a[i,2] <- res_cd8a$estimate
  result_cd8a[i,1] <- as.character(data.clean[i,1])
}
result_cd8a
write.csv(result_cd8a, file = "./correlation_result/result_cd8a.csv")
```

### Overlapping Analysis for correlation results

The strength of a correlation was measured by the correlation coefficient r. Correlation greater than 0.5 was used as cutoff. Seven biomarkers correlation genesets were used for overlapping analysis

**345 overlapping genes were selected out**,see the below dataframe

```{r overlapping_correlation, eval=T, echo=F}

## R >= 0.5
# HAVCR2
result.havcr2 <- read.csv("./correlation_result/result_havcr2.csv", header=TRUE, stringsAsFactors = FALSE)
havcr2_0.5 <- subset(result.havcr2, R >=0.5)
gene_list_havcr2_0.5 <- havcr2_0.5$variable

# LAG3
result_lag3 <- read.csv("./correlation_result/result_lag3.csv", header=TRUE, stringsAsFactors = FALSE)
lag3_0.5 <- subset(result_lag3, R >=0.5)
gene_list_lag3_0.5 <- lag3_0.5$variable

# PDCD1
result_pdcd1 <- read.csv("./correlation_result/result_pdcd1.csv", header=TRUE, stringsAsFactors = FALSE)
pdcd1_0.5 <- subset(result_pdcd1, R >=0.5)
gene_list_pdcd1_0.5 <- pdcd1_0.5$variable

# CD8A
result_cd8a <- read.csv("./correlation_result/result_cd8a.csv", header=TRUE, stringsAsFactors = FALSE)
cd8a_0.5 <- subset(result_cd8a, R >=0.5)
gene_list_cd8a_0.5 <- cd8a_0.5$variable

# EOMES
result_eomes <- read.csv("./correlation_result/result_eomes.csv", header=TRUE, stringsAsFactors = FALSE)
eomes_0.5 <- subset(result_eomes, R >=0.5)
gene_list_eomes_0.5 <- eomes_0.5$variable

# TBX21
result_tbx21 <- read.csv("./correlation_result/result_tbx21.csv", header=TRUE, stringsAsFactors = FALSE)
tbx21_0.5 <- subset(result_tbx21, R >=0.5)
gene_list_tbx21_0.5 <- tbx21_0.5$variable

# CTLA4
result_ctla4 <- read.csv("./correlation_result/result_ctla4.csv", header=TRUE, stringsAsFactors = FALSE)
ctla4_0.5 <- subset(result_ctla4, R >=0.5)
gene_list_ctla4_0.5 <- ctla4_0.5$variable

## ovelapping results
p.e.overlapping_0.5 <- Reduce(intersect, list(gene_list_havcr2_0.5,gene_list_lag3_0.5,gene_list_cd8a_0.5,gene_list_eomes_0.5, gene_list_tbx21_0.5))

p.e.overlapping_0.5
```

**Venn diagram showed the number of overlapping genes from five biomarkers correlated gene sets **

```{r venn_overlap_corre, eval=T, echo=F,out.width="60%"}

HAVCR2 <- sample(gene_list_havcr2_0.5, replace = FALSE)
LAG3 <- sample(gene_list_lag3_0.5, replace = FALSE)
CD8A <- sample(gene_list_cd8a_0.5, replace = FALSE)
EOMES <- sample(gene_list_eomes_0.5, replace = FALSE)
TBX21 <- sample(gene_list_tbx21_0.5, replace = FALSE)

venn.plot <- venn.diagram(
	x = list(
		HAVCR2 = HAVCR2,
		LAG3 = LAG3, 
		CD8A =CD8A,
		EOMES = EOMES,
		TBX21 =TBX21
		),
filename = "Venn_Tcell_Dysfunction_5genes_.png",
	col = "transparent",
	fill = 1:5,
	alpha = 0.50,
	cex = 1,
	fontfamily = "serif",
	fontface = "bold",
	cat.col = rep("black", 5),
	cat.cex = 1.3,
	cat.fontfamily = "serif",
	margin = 0.1
	);

knitr::include_graphics("Venn_Tcell_Dysfunction_5genes.png")
```

## Differential gene analysis

### Data preparation

Samples were ranked by overall survival time (os.time), and categorized into high survival group (75% percentile) and low survival group (25% percentile). 

Simplify samples name to "H1,H2,..H45" and "L1, L2,..L44"

```{r deg_analysis_data_prep, eval=T, echo=F}


# load survival datasets
p.s.data <- read.table(gzfile('./Data/TCGA-PAAD.survival.tsv.gz'), sep="\t", header=TRUE, stringsAsFactors=FALSE)

class(p.s.data$OS.time)
hi.p.s.data <- p.s.data[p.s.data$OS.time > quantile(p.s.data$OS.time, 0.75), ]
lo.p.s.data <- p.s.data[p.s.data$OS.time < quantile(p.s.data$OS.time, 0.25), ]

hi.p.s.data$sample <- substr(hi.p.s.data$sample, 1,15)
hi.samples.string.ID <- c("sample",hi.p.s.data$sample)

lo.p.s.data$sample <- substr(lo.p.s.data$sample, 1,15)
lo.samples.string.ID <- c("sample",lo.p.s.data$sample)

# load gene expression dataset
p.data <- read.table(gzfile('./Data/HiSeqV2.PADD.RSEM.normalized.gz'), sep="\t", heade=TRUE, stringsAsFactors=FALSE)

library(stringr)

# p.dataset
columnname <- str_replace_all(colnames(p.data), "[.]" , "-")
colnames(p.data) <- columnname

# subset high survival group
overlapping.name.h<- Reduce(intersect, list(columnname,hi.samples.string.ID))
p.data.hi <- p.data[, which(names(p.data) %in% overlapping.name.h)]
colnames(p.data.hi) <- c("Symbol", sprintf("H%d", 1:45))   # simplify sample name to "H1, H2...H45"

# subset low survival group

overlapping.name.l <- Reduce(intersect, list(columnname,lo.samples.string.ID))
p.data.low <- p.data[, which(names(p.data) %in% overlapping.name.l)]
colnames(p.data.low) <- c("Symbol",sprintf("L%d", 1:44)) # simplify sample name to "L1, L2...L44"

# Combine High survivor and Low survivor datasets together
data.s.h.l <- inner_join(p.data.hi, p.data.low, by = "Symbol")
row.names(data.s.h.l) <- data.s.h.l$Symbol
data.s.h.l <- data.s.h.l %>% dplyr::select(-c("Symbol")) 
data.s.h.l[1:6, ]
```

### Phenotypic analysis

Sample from High survival group "H4", "H15", "H18", "H23", "H25", "H26","H29","H40" were tumor free samples. They could be assumed as healthy specimen control.

```{r phenotype_prep, eval=T, echo=F}

# High survival group only keep "H1,H3,H5,H11,H31,H32,H36,H39"

selected.samples <- c("H4", "H15", "H18", "H23", "H25", "H26","H29","H40")

D1 <- data.s.h.l[, which(names(data.s.h.l) %in% selected.samples)]
selected.hi <- c(sprintf("H%d", 1:45))  
D2 <- data.s.h.l[, -which(names(data.s.h.l) %in% selected.hi)]
data.selected <- cbind(D1,D2)

```

### Comparison of high survival group vs. low survival group

```{r comparison, eval=T, echo=F}

snames <- colnames(data.selected)

factor1 <- substr(snames, 1, 1)
group <- interaction(factor1)

d0 <- DGEList(data.selected)
d0 <- calcNormFactors(d0)
d0$samples

cutoff <- 3
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d)

```
**1. plotMDS**

**MDS plot showed good separation between high survival group vs. low survival group**
```{r plotmds, eval=T, echo=T}
# plotMDS 
plotMDS(d, col = as.numeric(group))

mm <- model.matrix(~0 + group)
y <- voom(d, mm, plot = F)
```

**2. Fitting linear models in limma**

```{r deg_analysis_lima, eval=T, echo=F}
# Fitting linear models in limma
fit <- lmFit(y, mm)
head(coef(fit))

contr <- makeContrasts(groupH - groupL, levels = colnames(coef(fit)))
contr

tmp <- contrasts.fit(fit, contr)

tmp <- eBayes(tmp)
top.table <- topTable(tmp, sort.by = "P", n = Inf)
head(top.table, 30)

result.selected.fdr_0.5 <- top.table[which(top.table$adj.P.Val< 0.05), ]



```
Low survival correlated gene log(FC)<-0 and FDR <0.05, top 10 genes:
```{r deg_anno, eval=T, echo=F}

# Low survival correlated gene log(FC)<-0 and FDR <0.05

result.anno.low <- result.selected.fdr_0.5[which(result.selected.fdr_0.5$logFC< -0), ]
write.csv(result.anno.low, file = './DE_results/Selected_low_survivalcorrelated_padj0.05.csv')
result.anno.low[1:10,]
```

## Overlapping between correlation analysis and DEG analysis

Venn diagram showed 196 overlapping genes between DEG dataset and correlation dataset. 

```{r venn_correl_deg, eval=T, echo=F,out.width="60%"}

result.anno.low.05 <- read.csv("./DE_results/Selected_low_survivalcorrelated_padj0.05.csv", header=TRUE, stringsAsFactors = FALSE)
result.anno.low.05 <- result.anno.low.05[complete.cases(result.anno.low.05[,1]),]
correlation.data <-p.e.overlapping_0.5
DEG.data <- result.anno.low.05$X


overlapping.gene<- Reduce(intersect, list(correlation.data,DEG.data))

library("VennDiagram")
correlation_dataset <- sample(correlation.data, replace = FALSE)
DEG_dataset <- sample(DEG.data, replace = FALSE)


venn.plot2 <- venn.diagram(
	x = list(
		correlation_dataset= correlation_dataset,
		DEG_dataset = DEG_dataset 
		),
filename = "Venn_Correlation_DEG_.png",
	col = "transparent",
	fill = 2:3,
	alpha = 0.50,
	cex = 1,
	fontfamily = "serif",
	fontface = "bold",
	cat.col = rep("black", 2),
	cat.cex = 1.3,
	cat.fontfamily = "serif",
	margin = 0.1
	);

knitr::include_graphics("Venn_Correlation_DEG.png")

```

## GO and KEGG Analysis

196 overlapped genes from differential gene expression analysis and correlation anlysis were implemented by GO and KEGG analysis. Most of genes were classfied into T cell function, which gave an evidence that my data process could target to T cell level. 

```{r go_kegg_prep, eval=T, echo=F}
# GO and KEGG analysis
x <- overlapping.gene
test = bitr(x, 
fromType="SYMBOL", 
toType="ENTREZID",  
OrgDb="org.Hs.eg.db") 

ggo <- groupGO(gene = test$ENTREZID, OrgDb = org.Hs.eg.db, ont = "CC",level = 3,readable = TRUE)

data(geneList, package="DOSE") #
gene <- names(geneList)[abs(geneList) > 2]
gene.df <- bitr(gene, fromType = "ENTREZID", toType = c("ENSEMBL", "SYMBOL"), OrgDb = org.Hs.eg.db)
head(gene.df,2)


ego_ALL <- enrichGO(gene = test$ENTREZID, 
                universe = names(geneList),
                OrgDb = org.Hs.eg.db, 
                ont = "ALL", 
                pAdjustMethod = "BH",
                pvalueCutoff = 1, 
                qvalueCutoff = 1,
				readable = TRUE)
head(ego_ALL,2)
```

### Gene Ontology Molecular Function 


```{r go_mf, eval=T, echo=F, fig.height =7 , fig.width = 10, fig.align = "center"}
ego_MF <- enrichGO(gene = test$ENTREZID, universe = names(geneList),OrgDb = org.Hs.eg.db,ont = "MF", pAdjustMethod = "BH",pvalueCutoff = 1,qvalueCutoff = 1,readable = FALSE)
ego_MF1 <- setReadable(ego_MF, OrgDb = org.Hs.eg.db)

barplot(ego_MF, showCategory=20,title="EnrichmentGO_MF")
```

### Gene Ontology Biology Process

```{r go_bp, eval=T, echo=F, fig.height =7 , fig.width = 10, fig.align = "center"}
ego_BP <- enrichGO(gene = test$ENTREZID, universe = names(geneList),OrgDb = org.Hs.eg.db,ont = "BP", pAdjustMethod = "BH",pvalueCutoff = 1,qvalueCutoff = 1,readable = FALSE)
ego_BP1 <- setReadable(ego_BP, OrgDb = org.Hs.eg.db)

barplot(ego_BP, showCategory=20,title="EnrichmentGO_BP")
```

### Gene Ontology Cellular Component

```{r go_cc, eval=T, echo=F, fig.height =7 , fig.width = 10, fig.align = "center"}
ego_CC <- enrichGO(gene = test$ENTREZID, universe = names(geneList),OrgDb = org.Hs.eg.db,ont = "CC", pAdjustMethod = "BH",pvalueCutoff = 1,qvalueCutoff = 1,readable = FALSE)
ego_CC1 <- setReadable(ego_CC, OrgDb = org.Hs.eg.db)

barplot(ego_CC, showCategory=20,title="EnrichmentGO_CC")
```

### KEGG Pathway

```{r go_kegg, eval=T, echo=F, fig.height =7 , fig.width = 10, fig.align = "center"}

kk <- enrichKEGG(gene = test$ENTREZID,
                 organism = 'hsa', 
                 pvalueCutoff = 1)
head(kk)

dotplot(kk,title="Enrichment KEGG_dot")


hsa04659 <- pathview(gene.data = geneList,

pathway.id = "hsa04659", #

species = "hsa",

limit = list(gene=max(abs(geneList)), cpd=1))
```
## Correlation Results

### Heatmap

**Biomarkers: CD8A, HAVCR2, LAG3, PDCD1, EOMES, TBX21,CTLA4**

Example gene: **TNFRSF1B** was used to show the correlation result


```{r heatmap, eval=T, echo=F}

selected.gene <- c("CD8A", "HAVCR2", "LAG3", "PDCD1", "EOMES", "TBX21", "CTLA4", "ENTPD1", "CD244", "PRDM1", "TOX", "NR4A1","TNFRSF1B")
data.extract <- data.clean[data.clean$Symbol %in% selected.gene, ]
row.names(data.extract) <- data.extract$Symbol
data.extract <- data.extract[, -1]
data.extract.t <- data.frame(t(data.extract))
sample.name <- row.names(data.extract.t)


TNFRSF1B.hi <- data.extract.t[data.extract.t$TNFRSF1B >= quantile(data.extract.t$TNFRSF1B, 0.75), ]
TNFRSF1B.hi.name <- sprintf("TNFRSF1B_High%d", 1:46)
row.names(TNFRSF1B.hi) <- TNFRSF1B.hi.name
TNFRSF1B.hi.t <- data.frame(t(TNFRSF1B.hi))

TNFRSF1B.low <- data.extract.t[data.extract.t$TNFRSF1B <= quantile(data.extract.t$TNFRSF1B, 0.25), ]
TNFRSF1B.low.name <- sprintf("TNFRSF1B_Low%d", 1:46)
row.names(TNFRSF1B.low) <- TNFRSF1B.low.name
TNFRSF1B.low.t <- data.frame(t(TNFRSF1B.low))


TNFRSF1B.hi.low.merge <- merge(TNFRSF1B.hi.t, TNFRSF1B.low.t, by=0, all=TRUE)
row.names(TNFRSF1B.hi.low.merge) <- TNFRSF1B.hi.low.merge$Row.names
TNFRSF1B.hi.low.merge <- TNFRSF1B.hi.low.merge[, -1]
datawithoutTNFRSF1B <- TNFRSF1B.hi.low.merge[-12, ]


# Heatmap 
library(pheatmap)
library(RColorBrewer)
library(ggplot2)

mymat <- datawithoutTNFRSF1B
colnames(mymat) <- c(rep("H", 46), rep("L", 46))

mat_breaks <- seq(min(mymat), max(mymat), length.out = 10)

pheatmap(
  mat               = mymat,
  color             = viridis(length(mat_breaks) - 1),
  breaks            = mat_breaks,
  border_color      = NA,
  fontsize_col      = 5,
  show_colnames     = TRUE,
  show_rownames     = TRUE,
  cluster_cols      = FALSE,
  drop_levels       = TRUE,
  fontsize          = 14,
  main              = "TNFRSF1B"
)

```

Heatmap of a selected gene potentially effect T cell dysfunction in pancreatic patients from TCGA pancreatic cohort, exhibiting high (75th percentile) and low(25th percentile) TNFRSF1B expression


### Scatter plot

Scatter plot showed the positive correlation between TNFRSF1B with CTLA4.
```{r scatter_plot, eval=T, echo=F, fig.align = "center"}
# Scatter plot
 # Use PDCD1 as example
library(tidyverse)
library(dplyr)
data.extract.scatter <- data.clean[data.clean$Symbol %in% selected.gene, ]
row.names(data.extract.scatter) <- data.extract.scatter$Symbol
data.extract.scatter <- data.extract.scatter[, -1]
data.extract.scatter.t <- data.frame(t(data.extract.scatter))
sample.name <- row.names(data.extract.scatter.t)
data.scatter.tnfrsf1b.hacvr2 <- data.extract.scatter.t %>% dplyr::select(TNFRSF1B, CTLA4)

sp <- ggscatter(data.scatter.tnfrsf1b.hacvr2, x = "CTLA4", y = "TNFRSF1B",
   add = "reg.line",  # Add regressin line
   add.params = list(color = "blue", fill = "lightgray"), 
   conf.int = TRUE # Add confidence interval
   )
# Add correlation coefficient
sp + stat_cor(method = "spearman", label.x = 0, label.y = 8)
```

## Survival analysis
### Data prepration
```{r survival_analysis_dataprep, eval=T, echo=F, fig.align = "center"}
## Survival analysis

# Load transformed RSEM normalized count

data.rsem <- read.table(gzfile("./Data/HiSeqV2.PADD.RSEM.normalized.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)
columnname.r <- str_replace_all(colnames(data.rsem), "[.]" , "-")
colnames(data.rsem) <- columnname.r

# load survival datasets
data.survival <- read.table(gzfile("./Data/TCGA-PAAD.survival.tsv.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)

# Clean survival dataset
data.survival.1 <- data.survival[, -3]
data.survival.1$OS.time <- round((data.survival.1$OS.time)/30)

# Extract T cell dysfunction genes. Data comes from above overlapping analysis
data.extract.r <- data.rsem
data.extract.r <- data.rsem[data.rsem$sample %in% p.e.overlapping_0.5, ]
row.names(data.extract.r) <- data.extract.r$sample
data.extract.r <- data.extract.r[, -1]
data.extract.r.t <- data.frame(t(data.extract.r))
sample.name <- row.names(data.extract.r.t)
data.extract.r.t <- mutate(sample = as.character(sample.name), data.extract.r.t)

#Combine survival and extracted rsem gene datasets together

data.survival.1$sample <- substr(data.survival.1$sample, 1,15)
data.combine.r <- inner_join(data.survival.1,data.extract.r.t,by = "sample")
dim(data.combine.r)


library(tidyverse)
```

### Result

Kaplan–Meier curves were constructed by dividing specimens at the one standard deviation above or below average expression of each selected gene. 

TNFRSF1B was positively correlated with poor survival rate

```{r survival_analysis, eval=T, echo=F, fig.align = "center"}

Data.t.r <- scale(data.combine.r$TNFRSF1B,center = TRUE, scale = TRUE) # z score scaling
test.data.r <- data.frame(data.combine.r[,1:3], Data.t.r)
colnames(test.data.r) <- c('sample', 'OS', 'OS.time', 'TNFRSF1B')
highExpr <-1.0
lowEXPR <- -1.0
test.data.r$TNFRSF1B <- ifelse(test.data.r$TNFRSF1B >= highExpr, 'High',ifelse(test.data.r$TNFRSF1B <=lowEXPR, 'Low', 'Mid'))  # one standard deviation above average expression was defined as "High"

test.data.r$TNFRSF1B <- factor(test.data.r$TNFRSF1B,levels = c('Low', 'Mid', 'High'))

library(survival)
library(survminer)


ggsurvplot(survfit(Surv(OS.time, OS) ~ TNFRSF1B,
                   data = test.data.r),
           data = test.data.r,
           pval = TRUE,
           xlab = 'Month',
           break.time.by = 20,
           risk.table.y.text.col = TRUE,
           risk.table.y.text = FALSE)


```

## Summary

* Identified around two hundreds potential signatures of T cell dysfunction from pancreatic tumor cohort by testing how the expression of each gene interacted with the classical T cell specific dysfunction biomarkers. 

* These potential T cell dysfunction genes were highly correlated with T cell function and was correlated with poor survival rate.

* This workflow could be implemented into other TCGA cancer type cohorts to identify interesting genes.

## Limitation

* One limitation of this study is I only focused on TCGA tumor dataset, by using CAR T cell clinical trials dataset to further evaluate correlation results were necessary.

* Another limitation is that I focused primarily on gene expression biomarkers. However, other data types can also predict T cell dysfunction, like epigenetic and proteomic datasets.


