---
title: "BMIN503/EPID600 Final Project"
author: "Yufan Zhou"
output: 
  html_document:
    toc: false 
    depth: 3 
    theme: paper 
    highlight: tango
---
```{r set-options, echo=FALSE, cache=FALSE}
options(width = 400)
```  
***
Use this template to complete your project throughout the course. Your Final Project presentation in class will be based on the contents of this document. Replace the title/name and text below with your own, but keep the headers.

### Overview
Uncharacterized signatures of T cell dysfunction could be identified from tumor cohorts and mouse models data sets by testing how gene expression of putative signatures correlate with the specific classical T cell dysfunction biomarkers. Those candidates will be further validated by testing their correlation with CAR-T treatment outcomes from my in-house RNAseq dataset. Identification of reproducible biomarkers that can be applied to predict benefit of CAR-T therapy might be of clinical value.

### Introduction 
Chimeric antigen receptor (CAR) T cells have been genetically engineered to express a receptor that recognizes a specific antigen, have given rise to breakthroughs in treating hematological malignancies. However, their success in treating solid tumors has been limited. T cells that enter solid tumors can stop working due to a phenomenon called T cell dysfunction. T cell dysfunction is a hallmark of many cancers. Identifying and overcoming mechanisms associated with dysfunction in CAR T cells is of vital importance to generating CAR T cells that can proliferate and successfully eliminate tumor cells.

Gene expression biomarkers in T cells, such as PDCD1, HACVR2, LAG3, TOX, EOMES, CTLA4, CD8A have been demonstrated in mouse model to predict CAR-T treatment benefits. However, current CAR-T clinical trials have gene expression profiles on only a small number of samples, which are insufficient to train prognostic biomarkers. Alternatively, there are many public tumor profiling data sets from human clinical trials. Some tumors have high level of infiltration by cytotoxic T cells, and these T cells tend to be in a dysfunctional state. Analysis of TCGA data and GEO data might uncover T cell specific dysfunction related genes. Identification of reproducible biomarkers that can be applied to predict benefit of CAR-T therapy might be of clinical value
### Methods
Classical T cell specific dysfunction related genes were analyzed using the TCGA pancreatic cancer, prostate cancer and melanoma cohorts. TCGA genomic data were downloaded from the UCSC cancer genome browser project. Gene expression was measured experimentally using the Illumina HiSeq 2000 RNA Sequencing platform and log2(x + 1) transformed. Spearman’s rank correlation test will be used to assess the strength of the relationship between each classical T cell specific dysfunction reference gene with rest of 50,000 genes. Different correlation data sets will be performed overlapping analysis. For GEO mouse genomic datasets, standard RNAseq analysis workflow will be implemented. Overlapped genes from different gene data sets could be served as novel T cell dysfunction signatures. Finally, I will validate those T cell dysfunction signature candidates in my in-house CAR-T treatment patient data set.

### Results
Describe your results and include relevant tables, plots, and code/comments used to obtain them. End with a brief conclusion of your findings related to the question you set out to address. You can include references if you'd like, but this is not required.
