---
title: "BMIN503/EPID600 Final Project"
author: "Yufan Zhou"
output: 
  html_document:
    toc: false 
    depth: 3 
    theme: paper 
    highlight: tango
---
```{r set-options, echo=FALSE, cache=FALSE}
options(width = 400)
```  
***
### Overview
Adoptive cell therapy with chimeric antigen receptor expressing T (CAR T) cells has shown promising therapeutic efficacy against leukemia and lymphoma. However, CAR T cells in solid tumors fail to be as effective as in liquid tumors since they enter into a hyporesponsive (exhausted or dysfunctional) state that is induced by chronic antigen stimulation in cancer. To determine what kind of genes contribute to T cell exhaustion/ dysfunction from clinical patients sample would be meaningful to guide next-generation CAR T cell production. I identitified over two hundreds potential signatures of T cell dysfunction from pancreatic tumor cohort by testing how the expression of each gene in tumor interact with the classiscal T cell specific dysfunction biomarkers. These potential T cell dysfunction genes were highly correlated with T cell activation and was correlated with poor survival rate. 


### Introduction 
Chimeric antigen receptor (CAR) T cells have been genetically engineered to express a receptor that recognizes a specific antigen, have given rise to breakthroughs in treating hematological malignancies. However, their success in treating solid tumors has been limited. T cells that enter solid tumors can stop working due to a phenomenon called T cell dysfunction. T cell dysfunction is a hallmark of many cancers. Identifying and overcoming mechanisms associated with dysfunction in CAR T cells is of vital importance to generating CAR T cells that can proliferate and successfully eliminate tumor cells.

Gene expression biomarkers in T cells, such as PDCD1, HACVR2, LAG3, TOX, EOMES, CTLA4, CD8A have been demonstrated in mouse model to predict CAR T treatment benefits. However, current CAR T clinical trials have gene expression profiles on only a small number of samples, which are insufficient to uncover prognostic biomarkers. Alternatively, there are many public tumor profiling datasets from human clinical trials. Some specimens have high level of infiltration by cytotoxic T cells and poor survival rate, and T cells in these specimens tend to be in a dysfunctional state. Analyzing TCGA dataset might uncover uncharacterized T cell dysfunction genes. Identification of reproducible biomarkers that can be applied to enhance CAR T cell persistance.

### Methods

1. Data Source

Datasets:

Level Normalized FKPM counts (n=182) and survival data (n=222) were downloaded from GDC TCGA Pancreatic Cancer cohort 

Biomarkers:

T cell dysfunction biomarkers: CD8A, HAVCR2, LAG3, PDCD1, EOMES, TBX21, CTLA4


2. Correlation analysis

TCGA genomic data were downloaded from the UCSC cancer genome browser project. Gene expression was measured experimentally using the Illumina HiSeq 2000 RNA Sequencing platform and log2(x + 1) transformed. Spearman’s rank correlation test will be used to assess the strength of the relationship between each classical T cell biomarkers with rest of 50,000 genes. The strength of a correlation was measured by the correlation coefficient r. Correlation greater than 0.5 was used as criteria to select candidate gene. Biomarkers correlation gene pools were performed overlapping analysis. 

3. Survival analysis

Kaplan–Meier curves were constructed by dividing specimens at the one standard deviation above or below average expression of each selected  gene. Overall survival (OS) data were available for all 222 patients

4. Differential gene expression

Differential gene expression was performed with Limma. Limma-voom was used to identify transcripts that were significantly differntially expressed between experimental groups using adjusted P value of <0.05.


### Results

```{r}

### Intergrative correlation analysis of discovering potential T cell dysfunction realted genes from TCGA Pancreatic Cancer cohort

### Data Source

# Datasets
Normalized fkpm counts (n=182) and survival data (n=222) were downloaded from GDC TCGA Pancreatic Cancer cohort.

data download link: "https://xenabrowser.net/datapages/?cohort=GDC%20TCGA%20Pancreatic%20Cancer%20(PAAD)&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443"

# Biomarkers

T cell dysfunction biomarkers: CD8A, HAVCR2, LAG3, PDCD1, EOMES, TBX21, CTLA4


### Data Clean

# load datasets
data <- read.table(gzfile("./Data/TCGA-PAAD.htseq_fpkm.tsv.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)

# remove decimal of Ensembl_ID
data_modified <- sapply(strsplit(data$Ensembl_ID,"\\."), function(x) x[1])
data_new <- cbind(data_modified, data)
data_new <- data_new[, -2]
head(data_new)

# rename colume 1
library(dplyr)
data2 <- data_new %>% rename(ensembl_gene_id = data_modified)
head(data2)

# map annotation
library('biomaRt')
library("curl")
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
gene <-getBM(attributes=c("ensembl_gene_id","external_gene_name"), values =data2$ensembl_gene_id, mart = mart)
id <- match(data2$ensembl_gene_id , gene$ensembl_gene_id)
data2$Symbol <- gene$external_gene_name[id]
head(data2)

# Remove "NA" value in Symbol
data2 <- data2[complete.cases(data2[,184]),]
dim(data2)

# Clean the Data
library(dplyr)
data.clean <- cbind(symbol = data2$Symbol, data2)
data.clean <- subset(data.clean, select=-c(ensembl_gene_id,Symbol))
colnames(data.clean) <- c("Symbol",seq(1,182))
data.clean[complete.cases(data.clean), ]
```

```{r}

### T cell dysfunction biomarkers overlappping correlation analysis

T cell dysfunction biomarkers: HAVCR2, CD8A, LAG3, PDCD1, EOMES, TBX21, CTLA4

# Used HAVCR2 as example, "gene" could be replaced by above biomarkers respectively

# set an empty dataframe
result_gene <- data.frame((matrix(nrow = 56553, ncol = 2))) # gene is HAVCR2
colnames(result_gene) <- c("variable", "R")

# for loop to retreive R value

x <- data.frame(t(subset(data.clean, Symbol == "gene")[-1]))  # gene is HAVCR2
colnames(x) <- subset(data.clean, Symbol == "gene")[,1]

for (i in 1:56553){
  y = data.frame(t(subset(data.clean[i,])[-1]))
  colnames(y) <- subset(data.clean[i,])[,1]
  res_gene <- cor.test(x$gene, y[1:182,1], method = "spearman", exact=F) # gerne is HAVCR2
  result_gene[i,2] <- res_gene$estimate
  result_gene[i,1] <- as.character(data.clean[i,1])
}
result_gene # gene is HAVCR2

```

```{r}
### Overlapping Analysis

Used R >= 0.5 as cut-off

# HAVCR2
havcr2_0.5 <- subset(result_havcr2, R >=0.5)
gene_list_havcr2_0.5 <- havcr2_0.5$variable

# LAG3
lag3_0.5 <- subset(result_lag3, R >=0.5)
gene_list_lag3_0.5 <- lag3_0.5$variable

# PDCD1
pdcd1_0.5 <- subset(result_pdcd1, R >=0.5)
gene_list_pdcd1_0.5 <- pdcd1_0.5$variable

# CD8A
cd8a_0.5 <- subset(result_cd8a, R >=0.5)
gene_list_cd8a_0.5 <- cd8a_0.5$variable

# EOMES
eomes_0.5 <- subset(result_eomes, R >=0.5)
gene_list_eomes_0.5 <- eomes_0.5$variable

# TBX21
tbx21_0.5 <- subset(result_tbx21, R >=0.5)
gene_list_tbx21_0.5 <- tbx21_0.5$variable

# CTLA4
ctla4_0.5 <- subset(result_ctla4, R >=0.5)
gene_list_ctla4_0.5 <- ctla4_0.5$variable

## Ovelapping results
p.e.overlapping_0.5 <- Reduce(intersect, list(gene_list_havcr2_0.5,gene_list_lag3_0.5,gene_list_cd8a_0.5,gene_list_eomes_0.5, gene_list_tbx21_0.5))

# VennDiagram 
library("VennDiagram")
HAVCR2 <- sample(gene_list_havcr2_0.5, replace = FALSE)
LAG3 <- sample(gene_list_lag3_0.5, replace = FALSE)
CD8A <- sample(gene_list_cd8a_0.5, replace = FALSE)
EOMES <- sample(gene_list_eomes_0.5, replace = FALSE)
TBX21 <- sample(gene_list_tbx21_0.5, replace = FALSE)

venn.plot <- venn.diagram(
	x = list(
		HAVCR2 = HAVCR2,
		LAG3 = LAG3, 
		CD8A =CD8A,
		EOMES = EOMES,
		TBX21 =TBX21
		),
filename = "Venn_Tcell_Dysfuncion_8genes.tiff",
	col = "transparent",
	fill = 1:5,
	alpha = 0.50,
	cex = 1,
	fontfamily = "serif",
	fontface = "bold",
	cat.col = rep("black", 5),
	cat.cex = 1.3,
	cat.fontfamily = "serif",
	margin = 0.1
	);

```


```{r}
### Differential gene analysis
Samples were ranked by overall survival time (os.time), high (75% percentile) and low (25% percentile) os.time samples were selected for further analysis

# load survival datasets
p.s.data <- read.table(gzfile('./Data/TCGA-PAAD.survival.tsv.gz'), sep="\t", header=TRUE, stringsAsFactors=FALSE)

class(p.s.data$OS.time)
hi.p.s.data <- p.s.data[p.s.data$OS.time > quantile(p.s.data$OS.time, 0.75), ]
lo.p.s.data <- p.s.data[p.s.data$OS.time < quantile(p.s.data$OS.time, 0.25), ]

hi.samples.string.ID <- c("Ensembl_ID",hi.p.s.data$sample)
lo.samples.string.ID <- c("Ensembl_ID",lo.p.s.data$sample)

# load gene expression dataset
p.data <- read.table(gzfile('./Data/HiSeqV2.PADD.RSEM.normalized.gz'), sep="\t", heade=TRUE, stringsAsFactors=FALSE)

library(dplyr)
library(stringr)

# p.dataset
columnname <- str_replace_all(colnames(p.data), "[.]" , "-")
colnames(p.data) <- columnname

# subset high survivor group
overlapping.name.h<- Reduce(intersect, list(columnname,hi.samples.string.ID))
p.data.hi <- p.data[, which(names(p.data) %in% overlapping.name.h)]
colnames(p.data.hi) <- c("Symbol", sprintf("H%d", 1:45))   # simplify sample name to "H1, H2...H45"

# subset low survivor group

overlapping.name.l <- Reduce(intersect, list(columnname,lo.samples.string.ID))
p.data.low <- p.data[, which(names(p.data) %in% overlapping.name.l)]
colnames(p.data.low) <- c("Symbol",sprintf("L%d", 1:44)) # simplify sample name to "L1, L2...L44"

# Combine High survivor and Low survivor datasets together
data.s.h.l <- inner_join(p.data.hi, p.data.low, by = "Symbol")
row.names(data.s.h.l) <- data.s.h.l$Symbol
data.s.h.l <- data.s.h.l %>% dplyr::select(-c("Symbol")) 
```

```{r}
## lima DE analysis

library(edgeR)
library("RColorBrewer")

snames <- colnames(data.s.h.l)

factor1 <- substr(snames, 1, 1)
group <- interaction(factor1)

d0 <- DGEList(data.s.h.l)
d0 <- calcNormFactors(d0)
d0$samples

cutoff <- 3
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d)

# Use plotMDS check simmilarity/dis-similarity between High survival and low survival group
plotMDS(d,col = as.numeric(group))
```

```{r}
# Fitting linear models in limma

mm <- model.matrix(~0 + group)
y <- voom(d, mm, plot = T)

fit <- lmFit(y, mm)
head(coef(fit))

contr <- makeContrasts(groupH - groupL, levels = colnames(coef(fit)))
contr

tmp <- contrasts.fit(fit, contr)

tmp <- eBayes(tmp)
top.table <- topTable(tmp, sort.by = "P", n = Inf)
head(top.table, 30)

result.fdr_0.5 <- top.table[which(top.table$adj.P.Val< 0.05), ]

# Preliminary conclusion: sample from High survival group:  "H1","H3","H5","H11","H31","H32","H36","H39" were seperated from other high suvival samples. There were no statistical differences between High survial group vs Low survival group. After checking phenotype dataset, samples "H1","H3","H5","H11","H31","H32","H36","H39" were only pancreatic neuroendocrine tumor samples in total samples (n=222). They were much less common than exocrine tumors, but has better survival rate and therapy outcome(reference:https://www.cancer.org/cancer/pancreatic-neuroendocrine-tumor/about/what-is-pnet.html). PlotMDS results showed neuroendocrine tumor samples had different molecular profiles from exocrine tumor. Neuroendocrine tumor samples could be a negative T cell dysfunction control for further analysis.

```


```{r}
# Compared the molecular profiles difference between pancreatic neuroendocrine tumor samples with other pancreatic tumor samples which had low survival rate.

# High survival group only keep "H1,H3,H5,H11,H31,H32,H36,H39"

selected.samples <- c("H1", "H3", "H5", "H11", "H31", "H32","H36","H39")

D1 <- data.s.h.l[, which(names(data.s.h.l) %in% selected.samples)]
selected.hi <- c(sprintf("H%d", 1:45))  
D2 <- data.s.h.l[, -which(names(data.s.h.l) %in% selected.hi)]
data.selected <- cbind(D1,D2)

```

```{r}
# Differential Gene Analysis

snames <- colnames(data.selected)

factor1 <- substr(snames, 1, 1)
group <- interaction(factor1)

d0 <- DGEList(data.selected)
d0 <- calcNormFactors(d0)
d0$samples

cutoff <- 3
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d)

plotMDS(d, col = as.numeric(group))

mm <- model.matrix(~0 + group)
y <- voom(d, mm, plot = T)

# Fitting linear models in limma
fit <- lmFit(y, mm)
head(coef(fit))

contr <- makeContrasts(groupH - groupL, levels = colnames(coef(fit)))
contr

tmp <- contrasts.fit(fit, contr)

tmp <- eBayes(tmp)
top.table <- topTable(tmp, sort.by = "P", n = Inf)
head(top.table, 30)

result.selected.fdr_0.5 <- top.table[which(top.table$adj.P.Val< 0.05), ]
```


```{r}
# Annotation

# remove decimal of Ensembl_ID

data.modified <- sapply(strsplit(row.names(result.selected.fdr_0.5),"\\."), function(x) x[1])
data.new <- cbind(data.modified, result.selected.fdr_0.5)

# rename colume 1
library(dplyr)
result.anno <- data.new %>% dplyr::rename(ensembl_gene_id = data.modified)

# map annotation
library('biomaRt')
library("curl")
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
listAttributes(ensembl)
gene <-getBM(attributes=c("ensembl_gene_id","external_gene_name"), values =result.anno$ensembl_gene_id, mart = mart)
id <- match(result.anno$ensembl_gene_id , gene$ensembl_gene_id)
result.anno$Symbol <- gene$external_gene_name[id]
head(result.anno)


# Remove "NA" value in Symbol
result.anno <- result.anno[complete.cases(result.anno[,7]),]
dim(result.anno)

row.names(result.anno) <- result.anno$ensembl_gene_id
result.anno <- result.anno[, -1]

# select Low survival positively correlated gene by using log(FC)<-1 and FDR <0.05

result.anno.low <- result.anno[which(result.anno$logFC< -1), ]
write.csv(result.anno.low, file = './DE_results/Selected_survival_High_vs_Low_logfc_minus_1_padj0.05.csv')

```

```{r}
### Venngraph between correlation analysis and DEG analysis

correlation.data <-p.e.overlapping_0.5
DEG.data <- result.anno.low$Symbol

overlapping.gene<- Reduce(intersect, list(correlation.data,DEG.data))

library("VennDiagram")
correlation_dataset <- sample(correlation.data, replace = FALSE)
DEG_dataset <- sample(DEG.data, replace = FALSE)


venn.plot <- venn.diagram(
	x = list(
		correlation_dataset= correlation_dataset,
		DEG_dataset = DEG_dataset 
		),
filename = "Venn_Correlation_DEG.tiff",
	col = "transparent",
	fill = 2:3,
	alpha = 0.50,
	cex = 1,
	fontfamily = "serif",
	fontface = "bold",
	cat.col = rep("black", 2),
	cat.cex = 1.3,
	cat.fontfamily = "serif",
	margin = 0.1
	);

### There are 268 overlapping genes between DEG dataset and correlation dataset
```

```{r}

## GO and KEGG analysis

library(DOSE)
library(org.Hs.eg.db)
library(topGO)
library(clusterProfiler)
library(pathview)


x <- overlapping.gene
test = bitr(x, 
fromType="SYMBOL", 
toType="ENTREZID",  
OrgDb="org.Hs.eg.db") 

ggo <- groupGO(gene = test$ENTREZID, OrgDb = org.Hs.eg.db, ont = "CC",level = 3,readable = TRUE)

data(geneList, package="DOSE") #
gene <- names(geneList)[abs(geneList) > 2]
gene.df <- bitr(gene, fromType = "ENTREZID", toType = c("ENSEMBL", "SYMBOL"), OrgDb = org.Hs.eg.db)
head(gene.df,2)


ego_ALL <- enrichGO(gene = test$ENTREZID, 
                universe = names(geneList),
                OrgDb = org.Hs.eg.db, 
                ont = "ALL", 
                pAdjustMethod = "BH",
                pvalueCutoff = 1, 
                qvalueCutoff = 1,
				readable = TRUE)
head(ego_ALL,2)

# GO-MF
ego_MF <- enrichGO(gene = test$ENTREZID, universe = names(geneList),OrgDb = org.Hs.eg.db,ont = "MF", pAdjustMethod = "BH",pvalueCutoff = 1,qvalueCutoff = 1,readable = FALSE)
ego_MF1 <- setReadable(ego_MF, OrgDb = org.Hs.eg.db)

barplot(ego_MF, showCategory=20,title="EnrichmentGO_MF")


# GO-BP
ego_BP <- enrichGO(gene = test$ENTREZID, universe = names(geneList),OrgDb = org.Hs.eg.db,ont = "BP", pAdjustMethod = "BH",pvalueCutoff = 1,qvalueCutoff = 1,readable = FALSE)
ego_BP1 <- setReadable(ego_BP, OrgDb = org.Hs.eg.db)

barplot(ego_BP, showCategory=20,title="EnrichmentGO_BP")


# GO-CC
ego_CC <- enrichGO(gene = test$ENTREZID, universe = names(geneList),OrgDb = org.Hs.eg.db,ont = "CC", pAdjustMethod = "BH",pvalueCutoff = 1,qvalueCutoff = 1,readable = FALSE)
ego_CC1 <- setReadable(ego_CC, OrgDb = org.Hs.eg.db)

barplot(ego_CC, showCategory=20,title="EnrichmentGO_CC")


# KEGG Analysis
kk <- enrichKEGG(gene = test$ENTREZID,
                 organism = 'hsa', 
                 pvalueCutoff = 1)
head(kk)

dotplot(kk,title="Enrichment KEGG_dot")


hsa04659 <- pathview(gene.data = geneList,

pathway.id = "hsa04659", #

species = "hsa",

limit = list(gene=max(abs(geneList)), cpd=1))
```

```{r}

## Correlation Results

Heatmap of a selected gene potentially effect T cell dysfunction in pancreatic patients from TCGA pancreatic cohort, exhibiting high (75th percentile) and low(25th percentile) TNFRSF1B expression

# Subset T cell dysfunction gene. Data was from above overlapping analysis. Used TNFRSF1B as example
Biomarkers: CD8A, HAVCR2, LAG3, PDCD1, EOMES, TBX21,CTLA4,
Candidate gene: TNFRSF1B 

selected.gene <- c("CD8A", "HAVCR2", "LAG3", "PDCD1", "EOMES", "TBX21", "CTLA4", "ENTPD1", "CD244", "PRDM1", "TOX", "NR4A1","TNFRSF1B")
data.extract <- data.clean[data.clean$Symbol %in% selected.gene, ]
row.names(data.extract) <- data.extract$Symbol
data.extract <- data.extract[, -1]
data.extract.t <- data.frame(t(data.extract))
sample.name <- row.names(data.extract.t)

# exhibiting high (75th percentile) TNFRSF1B expression
TNFRSF1B.hi <- data.extract.t[data.extract.t$TNFRSF1B >= quantile(data.extract.t$TNFRSF1B, 0.75), ]
TNFRSF1B.hi.name <- sprintf("TNFRSF1B_High%d", 1:46)
row.names(TNFRSF1B.hi) <- TNFRSF1B.hi.name
TNFRSF1B.hi.t <- data.frame(t(TNFRSF1B.hi))

# exhibiting low (25th percentile) TNFRSF1B expression
TNFRSF1B.low <- data.extract.t[data.extract.t$TNFRSF1B <= quantile(data.extract.t$TNFRSF1B, 0.25), ]
TNFRSF1B.low.name <- sprintf("TNFRSF1B_Low%d", 1:46)
row.names(TNFRSF1B.low) <- TNFRSF1B.low.name
TNFRSF1B.low.t <- data.frame(t(TNFRSF1B.low))

# merge datasets
TNFRSF1B.hi.low.merge <- merge(TNFRSF1B.hi.t, TNFRSF1B.low.t, by=0, all=TRUE)
row.names(TNFRSF1B.hi.low.merge) <- TNFRSF1B.hi.low.merge$Row.names
TNFRSF1B.hi.low.merge <- TNFRSF1B.hi.low.merge[, -1]
datawithoutTNFRSF1B <- TNFRSF1B.hi.low.merge[-12, ]

# Heatmap 
library(pheatmap)
library(RColorBrewer)
library(gplots)

# use TNFRSF1B as example 

mymat <- datawithoutTNFRSF1B 
colnames(mymat) <- c(rep("High", 46), rep("Low", 46))

mat_breaks <- seq(min(mymat), max(mymat), length.out = 10)

pheatmap(
  mat               = mymat,
  color             = viridis(length(mat_breaks) - 1),
  breaks            = mat_breaks,
  border_color      = NA,
  show_colnames     = FALSE,
  show_rownames     = TRUE,
  cluster_cols      = FALSE,
  drop_levels       = TRUE,
  fontsize          = 14,
  main              = "TNFRSF1B"
)

```

```{r}
# Scatter plot

# used TNFRSF1B and PDCD1 as example

library(tidyverse)
library(dplyr)

data.extract.scatter <- data.clean[data.clean$Symbol %in% p.e.overlapping_0.5, ]
row.names(data.extract.scatter) <- data.extract.scatter$Symbol
data.extract.scatter <- data.extract.scatter[, -1]
data.extract.scatter.t <- data.frame(t(data.extract.scatter))
sample.name <- row.names(data.extract.scatter.t)
data.scatter.tnfrsf1b.pdcd1 <- data.extract.scatter.t %>% dplyr::select(TNFRSF1B, PDCD1) # TNFRSF1B, PDCD1 would changed to any other gene

sp <- ggscatter(data.scatter.tnfrsf1b.pdcd1, x = "PDCD1", y = "TNFRSF1B",
   add = "reg.line",  # Add regressin line
   add.params = list(color = "blue", fill = "lightgray"), 
   conf.int = TRUE # Add confidence interval
   )
# Add correlation coefficient
sp + stat_cor(method = "spearman", label.x = 0, label.y = 8)

```

```{r}
## Survival analysis

# Load transformed RSEM normalized count

data.rsem <- read.table(gzfile("./Data/HiSeqV2.PADD.RSEM.normalized.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)
columnname.r <- str_replace_all(colnames(data.rsem), "[.]" , "-")
colnames(data.rsem) <- columnname.r

# load survival datasets
data.survival <- read.table(gzfile("./Data/TCGA-PAAD.survival.tsv.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)

# Clean survival dataset
data.survival.1 <- data.survival[, -3]
data.survival.1$OS.time <- round((data.survival.1$OS.time)/30)

# Extract T cell dysfunction genes. Data comes from above overlapping analysis
data.extract.r <- data.rsem
data.extract.r <- data.rsem[data.rsem$sample %in% p.overlapping_0.5, ]
row.names(data.extract.r) <- data.extract.r$sample
data.extract.r <- data.extract.r[, -1]
data.extract.r.t <- data.frame(t(data.extract.r))
sample.name <- row.names(data.extract.r.t)
data.extract.r.t <- mutate(sample = as.character(sample.name), data.extract.r.t)

#Combine survival and extracted rsem gene datasets together

data.survival.1$sample <- substr(data.survival.1$sample, 1,15)
data.combine.r <- inner_join(data.survival.1,data.extract.r.t,by = "sample")
dim(data.combine.r)

# Kaplan–Meier curves were constructed by dividing specimens at the one standard deviation above or below average expression of each selected  gene

# use gene "TNFRSF1B" as an example

Data.t.r <- scale(data.combine.r$TNFRSF1B,center = TRUE, scale = TRUE) # z score scaling
test.data.r <- data.frame(data.combine.r[,1:3], Data.t.r)
colnames(test.data.r) <- c('sample', 'OS', 'OS.time', 'TNFRSF1B')
test.data.r$TNFRSF1B <- ifelse(test.data.r$TNFRSF1B> 1, 'High','Low') # one standard deviation above average expression was defined as "High"

library(survival)
library(survminer)
res.cox <- coxph(Surv(OS.time, OS) ~ TNFRSF1B, data = text.data)
pvalue <- summary(res.cox)$logtest["pvalue"]


ggsurvplot(survfit(Surv(OS.time, OS) ~ TNFRSF1B,
                   data = test.data.r),
           data = test.data.r,
           pval = TRUE,
           xlab = 'Month',
           break.time.by = 20,
           risk.table.y.text.col = TRUE,
           risk.table.y.text = FALSE)


```

### Conclusion

In summary, I explored the genomic and phenotypic pancreatic tumor datasets to identify 268 T cell dysfunction related genes. Most of genes were highly representative in other published T cell dysfuntion models, providing evidence that T cell biomarker-driven approach to identify correlation genes was available. Gene Ontology and Pathway analysis showed most of T cell dysfunction related genes were enriched in T cell activation, differentiation and proliferation module, which were consistant with our current knowledge. That aslo suggested when we knocked out T cell dysfunction genes to improve the T cell persistance, T cell initially growth might be impaired at the same time. More further validations need to be implemented by dry lab. Overall, this biomarker-driven approach had the ability to indentify potential biomarker related genes as well as might guide translational research to improve the CAR T cell performance. 

